src/NLog/Logger.cs

```diff
@@ -94,17 +97,70 @@ public bool IsEnabled(LogLevel level)
             return GetTargetsForLevel(level) != null;
         }
 
+        /// <summary>
+        /// Creates new logger that automatically appends the specified property to all log events (without changing current logger)
+        /// </summary>
+        /// <param name="propertyKey">Property Name</param>
+        /// <param name="propertyValue">Property Value</param>
+        /// <returns>New Logger object that automatically appends specified property</returns>
+        public Logger WithProperty(string propertyKey, object propertyValue)
+        {
+            if (string.IsNullOrEmpty(propertyKey))
+                throw new ArgumentException(nameof(propertyKey));
+
+            Logger newLogger = Factory.CreateNewLogger(GetType()) ?? new Logger();
+            newLogger.Initialize(Name, _configuration, Factory);
+            var contextProperties = _contextProperties != null
```

Comment:
please create a helper for creating contextProperties  - this is prevent to get the code complexity above the boundary

Reply:
Will fix it